### PR TITLE
Option: Additional Flutter Build params

### PR DESCRIPTION
**Pull request by viktorbenei**
New Option: Additional params for flutter build

These params will be passed to flutter build regardless of the platform.

Example: you can specify a build number for the build with the `--build-number` flutter build flag.

### CODE REVIEW

**Review by Bence1001 — COMMENTED** [review]
Code changes LGTM.

Could you please add a test (most likely an E2E test in `bitrise.yml`) that ensures that the new behavior works? Thanks!

I'll also ask for a review from the tech writer team for the new input.

File: step.yml [hunk 1]
```diff
@@ -73,6 +73,18 @@ inputs:
...
+  - additional_build_params: ""
+    opts:
+      title: "Additional params for flutter build"
+      summary: "Additional params for flutter build"
+      description: |-
+        Additional params for flutter build.
+        
+        Example: you can specify a Build Number for `flutter build`
+        via the `--build-number` flutter build param.
+        E.g. to set it to the `$BITRISE_BUILD_NUMBER` you can set this input
```
Comment (BanyikAnna) [inline step.yml:85]: E.g > For example,

File: step.yml [hunk 2]
```diff
@@ -73,6 +73,18 @@ inputs:
         - "both"
         - "ios"
         - "android"
+  - additional_build_params: ""
+    opts:
+      title: "Additional params for flutter build"
+      summary: "Additional params for flutter build"
```
Comment (Bence1001) [inline step.yml:79]: Let's capitalize "Flutter" here.

File: bitrise.yml [hunk 3]
```diff
@@ -114,37 +167,36 @@ workflows:
     - script:
         title: Clone sample app
         inputs:
-          - content: |-
-              git clone $SAMPLE_APP_URL .
+        - content: |-
```
Comment (godrei) [inline bitrise.yml:170]: pls use simple-git-clone step

File: step.yml [hunk 4]
```diff
@@ -73,6 +73,18 @@ inputs:
...
+      description: |-
+        Additional params for flutter build.
+        
+        Example: you can specify a Build Number for `flutter build`
```
Comment (godrei) [inline step.yml:83]: Note: wf editor presents this as a markdown, newlines could be kept by adding 2 spaces at the end of lines.

Comment (Bence1001) [reply step.yml:83]: In this case, there will be a line break after `Additional params for flutter build.`. Do you mean we should keep the empty line as well?

Comment (godrei) [reply step.yml:83]: IMO the empty line is fine, my comment applies to the content after the line break.

**Review by godrei — DISMISSED** [review]

**Review by godrei — DISMISSED** [review]

**Comment by Bence1001** [comment]
The build might have failed because rebuild doesn't work correctly with forked repos. I'm closing and reopening the PR to give it a try.

**Review by Bence1001 — APPROVED** [review]
Approved from my side, but I'll wait for @godrei's review as well.

**Review by godrei — APPROVED** [review]

**Pull request merged**